Description: Clear dlerror() just in case dlsym() legitimately returns
 NULL, to prevent misleading errors on Nexenta.  Thanks to Pavel Strashkin.

--- fakeroot-1.14.5.orig/libfakeroot.c
+++ fakeroot-1.14.5/libfakeroot.c
@@ -206,8 +206,10 @@ void load_library_symbols(void){
 
 #endif /* LIBFAKEROOT_DEBUGGING */
   for(i=0; next_wrap[i].doit; i++){
+ /* clear dlerror() just in case dlsym() legitimately returns NULL */
+    msg = dlerror();
     *(next_wrap[i].doit)=dlsym(get_libc(), next_wrap[i].name);
-    if ( (msg = dlerror()) != NULL){
+    if ( (next_wrap[i].doit == NULL) && (msg = dlerror()) != NULL){
       fprintf (stderr, "dlsym(%s): %s\n", next_wrap[i].name, msg);
 /*    abort ();*/
     }
